puppet/lib/puppet/util/puppetdb/command.rb

```diff
@@ -44,22 +56,26 @@ def initialize(command, version, certname, payload)
       # Puppet 4.1.0. We need a better answer to non-utf8 data end-to-end.
       }.to_pson, "Error encoding a '#{command}' command for host '#{certname}'")
     end
+    @command = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(command)
+    @version = version
+    @certname = Puppet::Util::Puppetdb::CharEncoding.coerce_to_utf8(certname)
```

Comment:
is this necessary on certname and command? need to ensure the checksums match but I'm not sure a certname can contain nonascii anyway